dnf-behave-tests/features/repoquery/main.feature

```diff
@@ -0,0 +1,467 @@
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      bottom-a1-1:2.0-1.noarch
+      bottom-a1-1:2.0-1.src
+      bottom-a2-1:1.0-1.src
+      bottom-a2-1:1.0-1.x86_64
+      bottom-a3-1:1.0-1.src
+      bottom-a3-1:1.0-1.x86_64
+      bottom-a3-1:2.0-1.src
+      bottom-a3-1:2.0-1.x86_64
+      broken-deps-1:1.0-1.src
+      broken-deps-1:1.0-1.x86_64
+      mid-a1-1:1.0-1.src
+      mid-a1-1:1.0-1.x86_64
+      mid-a2-1:1.0-1.src
+      mid-a2-1:1.0-1.x86_64
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME (nonexisting package)
+ When I execute dnf with args "repoquery dummy"
+ Then the exit code is 0
+  And stdout is empty
+
+Scenario: repoquery NAME
+ When I execute dnf with args "repoquery top-a"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-VERSION
+ When I execute dnf with args "repoquery top-a-2.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-VERSION-RELEASE
+ When I execute dnf with args "repoquery top-a-2.0-2"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-EPOCH:VERSION-RELEASE
+ When I execute dnf with args "repoquery top-a-2:2.0-2"
+ Then the exit code is 0
+  And stdout is
+      """
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-EPOCH:VERSION-RELEASE old epoch
+ When I execute dnf with args "repoquery top-a-1:2.0-2"
+ Then the exit code is 0
+  And stdout is empty
+
+Scenario: repoquery NAME NAME-EPOCH:VERSION-RELEASE
+ When I execute dnf with args "repoquery bottom-a1 top-a-2:2.0-2"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      bottom-a1-1:2.0-1.noarch
+      bottom-a1-1:2.0-1.src
+      top-a-2:2.0-2.src
+      top-a-2:2.0-2.x86_64
+      """
+
+Scenario: repoquery NAME-VERSION NAME-EPOCH:VERSION_GLOB-RELEASE
+ When I execute dnf with args "repoquery bottom-a1-1.0 top-a-1:[12].0-1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom-a1-1:1.0-1.noarch
+      bottom-a1-1:1.0-1.src
+      top-a-1:1.0-1.src
+      top-a-1:1.0-1.x86_64
+      top-a-1:2.0-1.src
+      top-a-1:2.0-1.x86_64
+      """
+
+# TODO almost the same as above (note the -), but this one fails
+@xfail
+Scenario: repoquery NAME-VERSION NAME-EPOCH:VERSION_GLOB2-RELEASE
+ When I execute dnf with args "repoquery bottom-a1-1.0 top-a-1:[1-2].0-1"
```

Comment:
However, this test is affected by the imperfect regex parsing. Here the dash in the regex causes the forms to also not match anything, and the last-resort full NEVRA match doesn't work either, because the glob argument is missing architecture and therefore is not full NEVRA.

Let's not let this particular issue hold back the PR though. I still want to investigate it a bit more and then I will create a bug for it so that we can work on it separately.

Reply:
Agreed. This will require more work, let's not block the PR on this.